clash-lib/src/Clash/Core/Evaluator.hs

```diff
@@ -297,6 +301,10 @@ unwind eval gbl tcm h k v = do
     Instantiate ty               -> return (instantiate h k' v ty)
     PrimApply nm ty tys vals tms -> primop eval gbl tcm h k' nm ty tys vals v tms
     Scrutinise _ alts            -> return (scrutinise h k' v alts)
+    CastTo ty1 ty2               -> return (castTo h k' v ty1 ty2)
```

Comment:
This is wrong.... what it should be doing e.g. in a cast from `Integer` to `Signed`, is to add the `Signed.fromInteger#` primitive! The reason being, that primitives operating on `Signed` are expecting to see this primitive, they're not expecting to see a cast.